std/math.d

```diff
@@ -1653,14 +1653,24 @@ real exp(real x) @trusted pure nothrow @nogc
         else
             static assert(0, "Not implemented for this architecture");
 
-        // Special cases.
-        // FIXME: set IEEE flags accordingly
+        // Special cases. Raises an overflow or underflow flag accordingly,
+        // except in the case for CTFE, where there are no hardware controls.
         if (isNaN(x))
             return x;
         if (x > OF)
-            return real.infinity;
+        {
+            if (__ctfe)
+                return real.infinity;
+            else
+                return real.max * copysign(real.max, real.infinity);
```

Comment:
I answered this last year.

> This operation is deliberate because I didn't want the optimizer to fold the operation so that no hardware flag is raised.

Essentially, I *want* runtime to execute this so that the hardware flag is raised. If CTFE does the job, it returns the answer with no hardware exception.  Which defies the whole point of this PR.

If you have any alternatives, I'm all ears.